lib/gui/pages/settings/controllers/settings.js

```diff
@@ -53,72 +53,48 @@ module.exports = function(WarningModalService, SettingsModel, AnalyticsService)
   this.model = SettingsModel;
 
   /**
-   * @summary Enable a dangerous setting
+   * @summary Toggle setting
    * @function
    * @public
    *
-   * @param {String} name - setting name
-   * @param {Object} options - options
-   * @param {String} options.description - modal description
-   * @param {String} options.confirmationLabel - modal confirmation label
+   * @description
+   * If warningOption is given, it should be an object having `description` and `confirmationLabel`;
+   * these will be used to present a user confirmation modal before enabling the setting.
+   *
+   * @param {String} setting - setting key
+   * @param {Object} warningOptions - (optional) WarningModalService options
    * @returns {Undefined}
    *
    * @example
-   * SettingsController.enableDangerousSetting('unsafeMode', {
+   * SettingsController.toggle('unsafeMode', {
    *   description: 'Don\'t do this!',
    *   confirmationLabel: 'Do it!'
    * });
    */
-  this.enableDangerousSetting = (name, options) => {
-    if (!this.currentData[name]) {
+  this.toggle = (setting, warningOptions) => {
+
+    const value = this.currentData[setting];
+    const dangerous = Boolean(warningOptions);
 
-      AnalyticsService.logEvent('Set setting', {
-        setting: name,
-        value: false,
-        dangerous: false
-      });
+    AnalyticsService.logEvent('Toggle setting', {
+      setting,
+      value,
+      dangerous
+    });
```

Comment:
Ahh. Would needing to check for two events (which setting was being changed, and whether the warning was accepted or rejected) make the analytics harder to analyse though?